lib/elements/number.js

```diff
@@ -1,39 +1,78 @@
 const color = require('clorox');
 const Prompt = require('./prompt');
 const { cursor, erase } = require('sisteransi');
-const { style: stl, clear } = require('../util');
+const { style, clear } = require('../util');
 
 const isNumber = /[0-9]/;
+const isValidChar = /\.|-/;
 const isDef = any => any !== undefined;
-const isNull = any => any === null;
+const round = (number, precision) => {
+  let factor = Math.pow(10, precision);
+  return Math.round(number * factor) / factor;
+}
 
+/**
+ * NumberPrompt Base Element
+ * @param {Object} opts Options
+ * @param {String} opts.message Message
+ * @param {String} [opts.style='default'] Render style
+ * @param {Number} [opts.initial] Default value
+ * @param {Number} [opts.max=+Infinity] Max value
+ * @param {Number} [opts.min=-Infinity] Min value
+ * @param {Boolean} [opts.float=false] Parse input as floats
+ * @param {Number} [opts.round=2] Round floats to x decimals
+ * @param {Number} [opts.increment=1] Number to increment by when using arrow-keys
+ */
 class NumberPrompt extends Prompt {
-  constructor({ message, initial = '', min, max, style = 'default' }) {
-    super();
-
-    this.msg = message;
-    this.transform = stl.render(style);
-
-    this.min = isDef(min) ? min : -Infinity;
-    this.max = isDef(max) ? max : Infinity;
-    this.value = initial;
-
+  constructor(opts={}) {
+    super(opts);
+    this.transform = style.render(opts.style);
+    this.msg = opts.message;
+    this.initial = isDef(opts.initial) ? opts.initial : '';
+    this.float = !!opts.float;
+    this.round = opts.round || 2;
+    this.inc = opts.increment || 1;
+    this.min = isDef(opts.min) ? opts.min : -Infinity;
+    this.max = isDef(opts.max) ? opts.max : Infinity;
+    this.value = ''
     this.typed = '';
     this.lastHit = 0;
+    this.render();
+  }
 
-    this.initialValue = this.value;
+  set value(v) {
+    if (!v && v !== 0) {
```

Comment:
This check looks kinda confusing at first. I think you're just trying to show the placeholder if passed a `null` value, right?

Might be better off with `if (v == null)` which will match `null` and `undefined` but exclude `0`.

Reply:
If any false values except `0` are passed that code should run. Can I use `v == null` for that?

Reply:
Oh right, derp. Disregard then~!